### PR TITLE
Add prepayments to front payment display

### LINKED ISSUE #6109: Display on the front payments screen all prepayments
### Describe the problem

There is no easy access to prepayments in the system. 

### Describe the solution you'd like

when the receptionist charges the patient, she can see that there is a pre-payment and apply it as needed

### PR DESCRIPTION
**Pull request by juggernautsei**
Fixes #6109 

#### Short description of what this resolves:
Show any prepayments made by the patient. 

#### Changes proposed in this pull request:
add a function to get prepayments then display that on the front payment screen.

### CODE REVIEW

File: interface/patient_file/front_payment.php [hunk 1]
```diff
@@ -1151,6 +1151,17 @@ function make_insurance() {
                     <input name='form_pid' type='hidden' value='<?php echo attr($pid) ?>' />
                     <fieldset>
                         <legend><?php echo xlt('Payment'); ?></legend>
+                        <?php //ALB Showing unallocated pre-payments here
+                            if (get_unallocated_patient_balance($pid) > 0) : ?>
+                                <div class="col-12 oe-custom-line">
+                                    <label class="control-label" for="unallocated"><?php
+                                                echo xlt('Patient has an unallocated pre-payment amount of ') .
+                                                    get_unallocated_patient_balance($pid); ?></label><br>
```
Comment (stephenwaite) [inline interface/patient_file/front_payment.php:1159]: think you need this line to be
```
xlt(get_unallocated_patient_balance($pid)); ?></label><br>

File: interface/patient_file/front_payment.php [hunk 2]
```diff
@@ -1151,6 +1151,17 @@ function make_insurance() {
...
+                                    <a href="../billing/edit_payment.php?payment_id=<?php
+                                        echo xla(get_unallocated_payment_id($pid)); ?>"
```
Comment (stephenwaite) [inline interface/patient_file/front_payment.php:1161]: think you need this line to be
```
echo attr_url(get_unallocated_payment_id($pid)); ?>"

File: interface/patient_file/front_payment.php [hunk 3]
```diff
@@ -1151,6 +1151,17 @@ function make_insurance() {
                     <input name='form_pid' type='hidden' value='<?php echo attr($pid) ?>' />
                     <fieldset>
                         <legend><?php echo xlt('Payment'); ?></legend>
+                        <?php //ALB Showing unallocated pre-payments here
+                            if (get_unallocated_patient_balance($pid) > 0) : ?>
+                                <div class="col-12 oe-custom-line">
+                                    <label class="control-label" for="unallocated"><?php
+                                                echo xlt('Patient has an unallocated pre-payment amount of ') .
+                                                    xlt(get_unallocated_patient_balance($pid)); ?></label><br>
```
Comment (bradymiller) [inline interface/patient_file/front_payment.php:1159]: just need to escape above line (not translate it)
```
text(get_unallocated_patient_balance($pid)); ?></label><br>
```

File: library/patient.inc.php [hunk 4]
```diff
@@ -1822,3 +1822,17 @@ function updateDupScore($pid)
     );
     return $dupscore;
 }
+
+function get_unallocated_payment_id($pid)
```
Comment (bradymiller) [inline library/patient.inc.php:1826]: where is the `get_unallocated_patient_balance()` function which is used in above script?
(this is different where is `get_unallocated_payment_id()`)

Comment (bradymiller) [reply library/patient.inc.php:1826]: Ok, I found the `get_unallocated_patient_balance()` function which is already in the codebase.
So no my question is just what this `get_unallocated_payment_id()` function is needed for?

Comment (stephenwaite) [reply library/patient.inc.php:1826]: just brought in e1b2a591163f30daba16248266e31f013083ee0f

**Comment by bradymiller** [comment]
review complete

File: library/patient.inc.php (hunk 4 above)
Comment (bradymiller) [reply library/patient.inc.php:1826]: also just noted where get_unallocated_payment_id() is used above :)

**Comment by bradymiller** [comment]
If testing well and the styling issue (looks like just one spot) is addressed, then this is ready to come into the codebase.

2DV

**Comment by stephenwaite** [comment]
southpark_cheering

**Pull request merged**